src/coreclr/src/tools/Common/TypeSystem/Common/MetadataFieldLayoutAlgorithm.cs

```diff
@@ -452,13 +450,19 @@ protected ComputedInstanceFieldLayout ComputeAutoFieldLayout(MetadataType type,
             int instanceValueClassFieldCount = 0;
             int instanceGCPointerFieldsCount = 0;
             int[] instanceNonGCPointerFieldsCount = new int[maxLog2Size + 1];
+            LayoutInt largestAlignmentRequired = LayoutInt.One;
 
             foreach (var field in type.GetFields())
             {
                 if (field.IsStatic)
                     continue;
 
                 TypeDesc fieldType = field.FieldType;
+
+                var fieldSizeAndAlignment = ComputeFieldSizeAndAlignment(fieldType, packingSize);
```

Comment:
Could you please do this in the `// Iterate over all fields and do the following` foreach loop?

That loop already calls ComputeFieldSizeAndAlignment in one of the branches and we can save us a call.

Reply:
oh, I didn't notice that the comment addresses specified place in code and picked the first loop of in the body